lib/Analysis/ControlDependenceAnalysis.cpp

```diff
@@ -0,0 +1,333 @@
+//===- ControlDependenceAnalysis.h - Control dependence analyis utilities
+//----------*- C++ -*-===//
+//
+// Dynamatic is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Declares some utility functions useful to analyzing the control dependencies
+// between basic blocks of the CFG. The control dependendies are calculated
+// using the algorithm from the following paper
+//   J.Ferrante, K.J. Ottenstein, and J. D. Warren, "The Program Dependence
+//   Graph and its Use in Optimizations", ACM Trans. Program. Lang. Syst., vol.
+//   9, pp. 319-349, 1987.
+//
+//===----------------------------------------------------------------------===//
+
+#include "dynamatic/Analysis/ControlDependenceAnalysis.h"
+#include "llvm/Support/Debug.h"
+
+using namespace mlir;
+using namespace llvm;
+using namespace dynamatic;
+
+#define DEBUG_TYPE "cdg-analysis"
+
+void ControlDependenceAnalysis::identifyAllControlDeps(
+    mlir::func::FuncOp &funcOp) {
```

Comment:
I think we could drop the "mlir" identifier here if we are already using it in the namespace